walker/tar.go

```diff
@@ -91,3 +95,67 @@ func underSkippedDir(filePath string, skipDirs []string) bool {
 	}
 	return false
 }
+
+// fileWithTarOpener opens a file in a Tar.
+// If the file size is greater than or equal to ThresholdSize(200MB), it writes the file and caches the file name.
+// If the file size is less than ThresholdSize(200MB), it opens the file once and the content is shared so that some analyzers can use the same data
+func (w *walker) fileWithTarOpener(fi os.FileInfo, r io.Reader) func() (io.ReadCloser, func() error, error) {
+
+	var once sync.Once
+	var b []byte
+	var tempFilePath string
+	var tempDirPath string
+	var err error
+
+	return func() (io.ReadCloser, func() error, error) {
+		once.Do(func() {
+			if fi.Size() >= ThresholdSize {
+				var f *os.File
+				tempDirPath, err = ioutil.TempDir("", "trivy-*")
+				if err != nil {
+					err = xerrors.Errorf("failed to create the temp dir: %w", err)
+					return
+				}
```

Comment:
Any reason to create a temp directory?  Looks like `os.CreateTemp` is enough here.